runtime/doc/lua.txt

```diff
@@ -1436,6 +1436,36 @@ notify_once({msg}, {level}, {opts})                        *vim.notify_once()*
     Return: ~
         (boolean) true if message was displayed, else false
 
+on_fun({container}, {key}, {fn})                                *vim.on_fun()*
+    Sets `container[key]` to a new function that wraps both `fn` and the
+    original function `container[key]`. If the wrapper is called it invokes
+    `fn` with the arguments expected by the original function; if `fn` does
+    not return `false`, the wrapper then invokes the original function.
```

Comment:
I'm surprised that I couldn't find something like this in https://github.com/rxi/lume or https://github.com/luafun/luafun .

Does anyone have a util function like this in their plugins?